CRM/Contribute/BAO/Contribution.php

```diff
@@ -4684,6 +4694,7 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
       civicrm_api3('Contribution', 'sendconfirmation', array(
         'id' => $contribution->id,
         'payment_processor_id' => $paymentProcessorId,
+        'is_email_receipt' => $values['is_email_receipt'],
```

Comment:
I think the test failures are related to undefined index `is_email_receipt` here. Can you use `CRM_Utils_Array::value('is_email_receipt', $values )` and check ?